docs/source/user-guide/supported_models.rst

```diff
@@ -0,0 +1,65 @@
+.. _supported_models:
+
+====================
+Supported Models
+====================
+
+Similar to scikit-learn API, a model will be compatible with deepchecks' checks if it implements some minimal functionality.
+Without a compatible model, deepchecks can't run all checks that require predictions, for example all :docs:`performance checks </api/checks/performance>`.
+
+Below are the minimal requirements for each task type:
+
+Required API
+=============
+
+Regression
+----------
+
+All that is necessary for a regression model is the ``predict`` function.
+The predict function should expect an `array-like` or `sparse matrix` of shape ``(n_samples, n_features)``
+and is expected to return an `ndarray` of shape ``(n_samples,)``, a vector containing the predicted value for each sample.
+
+
+Example of a simple model:
+
+>>> class simple_regression_model:
+...     def predict(X: pd.DataFrame) -> pd.Series:
+...         ...
+
+Classification
+--------------
+
+For classification models, we require also the ``predict_proba`` function.
+The predict function should expect an `array-like` or `sparse matrix` of shape ``(n_samples, n_features)``
+and is expected to return an `ndarray` of shape ``(n_samples,n_classes)``, an array containing the predicted probability of each class per sample.
+
+
+
+>>> class simple_classification_model:
+...     def predict(X: pd.DataFrame) -> pd.Series:
+...         ...
+...     def predict_proba(X: pd.DataFrame) -> pd.Series:
+...         ...
+
+
+Other Helpful APIs
+==================
+
+Feature importance
+-------------------
+
+Deepchecks can calculate feature importance using `sklearn permutational_importance <https://scikit-learn.org/stable/modules/permutation_importance.html>`_.
```

Comment:
The explanation is good, but I'm missing - what is the default behavior...
"it can  do this and it can support that - but what actually happens"?

and - where is the feature importance used?
and how can I get it if I want?